### PR TITLE
✨ Add azure.monitor.diagnosticSettings and azure.monitor.activitylog

### PR DESCRIPTION
**Pull request by preslavgerchev**
- `azure.monitor.diagnosticSettings` to fetch subscription diagnostic settings
 - `azure.monitor.activitylog.alerts` to fetch alerts for the subscription

### CODE REVIEW

File: resources/packs/azure/azure.lr [hunk 1]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
 azure.monitor {
   // List of log profiles
   logProfiles() []azure.monitor.logprofile
+  // List of diagnostic settings for the subscription
+  diagnosticSettings() []azure.monitor.diagnosticsetting
+  // the activity log for the monitor
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Activity log for the monitor
```

File: resources/packs/azure/azure.lr [hunk 2]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+}
+
+// Azure Monitor Activity Log
+private azure.monitor.activitylog {
+  alerts() []azure.monitor.activitylog.alert
 }
 
+// Azure Monitor Activity Log Alert
+private azure.monitor.activitylog.alert {
+  // the identifier of the activity log alert
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // ID of the activity log alert
```

File: resources/packs/azure/azure.lr [hunk 3]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  id string
+  // the type of the activity log alert
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Type of the activity log alert
```

File: resources/packs/azure/azure.lr [hunk 4]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  type string
+  // the name of the activity log alert
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Name of the activity log alert
```

File: resources/packs/azure/azure.lr [hunk 5]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  name string
+  // the description of the activity log alert
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Description of the activity log alert
```

File: resources/packs/azure/azure.lr [hunk 6]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  description string
+  // the conditions for the activity log alert, all of these conditions must be met
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Conditions for the activity log alert, all of which must be met
```

File: resources/packs/azure/azure.lr [hunk 7]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  conditions []azure.monitor.activitylog.alert.condition
+  // the location of the alert
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Location of the alert
```

File: resources/packs/azure/azure.lr [hunk 8]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  location string
+  // the tags of the alert
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Tags of the alert
```

File: resources/packs/azure/azure.lr [hunk 9]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  tags map[string]string
+  // the actions that will activate when the conditions are met
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Actions that activate when the conditions are met
```

File: resources/packs/azure/azure.lr [hunk 10]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  actions []azure.monitor.activitylog.alert.action
+  // a list of resource IDs that will be used as prefixes, the alert applies to events where the resources id has any of the prefixes.
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // List of resource IDs that must be present to trigger the alert
```

File: resources/packs/azure/azure.lr [hunk 11]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  scopes []string
+}
+
+// Azure Monitor Activity Log Alert Condition
+ private azure.monitor.activitylog.alert.condition {
+  // the identifier of the condition
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // ID of the condition
```

File: resources/packs/azure/azure.lr [hunk 12]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  id string
+  // the alert's value will be compared to this value to determine if the condition is met. exclusive with 'containsAny'
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Value that is compared to the alert value to determine if the condition is met (exclusive with 'containsAny')
```

File: resources/packs/azure/azure.lr [hunk 13]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  equals string
+  // the name of the activity log alert's field that this condition will examine.
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Name of the field in the activity log alert that this condition examines
```

File: resources/packs/azure/azure.lr [hunk 14]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  fieldName string
+  // the alert's value will be compared to the values in this array to determine if the condition is met. exclusive with 'equals'
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Array of values that are compared to the alert's value to determine if the condition is met (exclusive with 'equals')
```

File: resources/packs/azure/azure.lr [hunk 15]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  containsAny []string
+  // the alert value will be compared against the conditions in this array. exclusive with all other fields.
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Array of conditions compared to the alert value to determine if the condition is met (exclusive with all other fields)
```

File: resources/packs/azure/azure.lr [hunk 16]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  anyOf []azure.monitor.activitylog.alert.condition
+ }
+
+
+// Azure Monitor Activity Log Alert Action
+ private azure.monitor.activitylog.alert.action {
+  // the identifier of the condition
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // ID of the condition
```

File: resources/packs/azure/azure.lr [hunk 17]
```diff
@@ -842,8 +842,62 @@ private azure.keyvault.secret  @defaults("id secretName") {
...
+  actionGroupId string
+  // the dictionary of custom properties to include with the post operation. These data are appended to the webhook payload.
```
Comment (misterpantz) [inline resources/packs/azure/azure.lr]:
```suggestion
  // Dictionary of custom properties to include with the post operation. These are appended to the webhook payload.
```

**Review by misterpantz — COMMENTED** [review]
Super, thanks for the opportunity to review!

File: resources/packs/azure/azure_rm_monitor.go [hunk 18]
```diff
@@ -73,10 +74,122 @@ func (a *mqlAzureMonitor) GetLogProfiles() ([]interface{}, error) {
...
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, err
+		}
+		for _, entry := range page.Value {
+			actions := []interface{}{}
+			conditions := []interface{}{}
+
+			for _, act := range entry.Properties.Actions.ActionGroups {
```
Comment (chris-rock) [inline resources/packs/azure/azure_rm_monitor.go:142]: I think we can simplify this by designing Action groups and conditions as dicts

**Review by chris-rock — APPROVED** [review]
Thank you @preslavgerchev

**Pull request merged**